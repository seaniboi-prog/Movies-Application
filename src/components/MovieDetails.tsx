import React from 'react'
import { Movie } from '../models';
import './detailsPage.scss';
import { MdOutlineAccessTimeFilled, MdCategory} from 'react-icons/md'
import { ImStarFull, ImStarHalf, ImStarEmpty } from 'react-icons/im'

interface Props {
    chosenMovie:Movie | undefined;
}

const MovieDetails:React.FC<Props> = ({ chosenMovie } : Props) => {

    const renderStars = (rating: string | undefined) => {

        const emptyStar:JSX.Element = <ImStarEmpty className='detail__icon'/>
        const halfStar:JSX.Element = <ImStarHalf className='detail__icon'/>
        const fullStar:JSX.Element = <ImStarFull className='detail__icon'/>

        switch(Math.round(Number(rating))) {
            case 0:
                return (<>{emptyStar} {emptyStar} {emptyStar} {emptyStar} {emptyStar}</>);
            case 1:
                return (<>{halfStar} {emptyStar} {emptyStar} {emptyStar} {emptyStar}</>);
            case 2:
                return (<>{fullStar} {emptyStar} {emptyStar} {emptyStar} {emptyStar}</>);
            case 3:
                return (<>{fullStar} {halfStar} {emptyStar} {emptyStar} {emptyStar}</>);
            case 4:
                return (<>{fullStar} {fullStar} {emptyStar} {emptyStar} {emptyStar}</>);
            case 5:
                return (<>{fullStar} {fullStar} {halfStar} {emptyStar} {emptyStar}</>);
            case 6:
                return (<>{fullStar} {fullStar} {fullStar} {emptyStar} {emptyStar}</>);
            case 7:
                return (<>{fullStar} {fullStar} {fullStar} {halfStar} {emptyStar}</>);
            case 8:
                return (<>{fullStar} {fullStar} {fullStar} {fullStar} {emptyStar}</>);
            case 9:
                return (<>{fullStar} {fullStar} {fullStar} {fullStar} {halfStar}</>);
            case 10:
                return (<>{fullStar} {fullStar} {fullStar} {fullStar} {fullStar}</>);
            case undefined:
                return (<>{emptyStar} {emptyStar} {emptyStar} {emptyStar} {emptyStar}</>);
            default:
                return (<>{emptyStar} {emptyStar} {emptyStar} {emptyStar} {emptyStar}</>);

        }
    }

    return (
    <div className='movie__info__box' data-testid='infoBox'>

        <div className='movie__info__line'>
            <p data-testid='infoDesc'>{chosenMovie?.description}</p>
        </div>

        <div className='movie__info__rating'>
            {/* <AiFillStar className='detail__icon'/> */}
            {/* <div data-testid='infoRating'>{chosenMovie?.rate}</div> */}
            {renderStars(chosenMovie?.rate)}
        </div>

        <div className='movie__info__line'>
            <MdOutlineAccessTimeFilled className='detail__icon'/>
            <div data-testid='infoLength'>{chosenMovie?.length}</div>
        </div>

        <div className='movie__info__line'>
            <MdCategory className='detail__icon'/>
            <div data-testid='infoGenre'>{chosenMovie?.genres.map(g => g[0].toUpperCase() + g.slice(1)).join(', ')}</div>
        </div>
    </div>
    );
}

export default MovieDetails