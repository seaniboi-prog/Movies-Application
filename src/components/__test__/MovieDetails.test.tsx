import React from 'react'
import ReactDOM from 'react-dom'
import MovieDetails from '../MovieDetails'

import { render, cleanup } from '@testing-library/react'
import 'jest-dom/extend-expect'
import renderer from 'react-test-renderer'