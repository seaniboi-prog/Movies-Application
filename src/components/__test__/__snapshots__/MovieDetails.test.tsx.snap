// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Details Component Tests Matches Snapshot 1`] = `
<div
  className="movie__info__box"
  data-testid="infoBox"
>
  <div
    className="movie__info__line"
    data-testid="infoLine"
  >
    <p
      data-testid="infoDesc"
    >
      description
    </p>
  </div>
  <div
    className="movie__info__line"
    data-testid="infoLine"
    id="movie__info_rating"
  >
    <svg
      className="detail__icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      version="1.1"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 6.204l-5.528-0.803-2.472-5.009-2.472 5.009-5.528 0.803 4 3.899-0.944 5.505 4.944-2.599 4.944 2.599-0.944-5.505 4-3.899zM8 11.773l-3.492 1.836 0.667-3.888-2.825-2.753 3.904-0.567 1.746-3.537 1.746 3.537 3.904 0.567-2.825 2.753 0.667 3.888-3.492-1.836z"
      />
    </svg>
     
    <svg
      className="detail__icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      version="1.1"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 6.204l-5.528-0.803-2.472-5.009-2.472 5.009-5.528 0.803 4 3.899-0.944 5.505 4.944-2.599 4.944 2.599-0.944-5.505 4-3.899zM8 11.773l-3.492 1.836 0.667-3.888-2.825-2.753 3.904-0.567 1.746-3.537 1.746 3.537 3.904 0.567-2.825 2.753 0.667 3.888-3.492-1.836z"
      />
    </svg>
     
    <svg
      className="detail__icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      version="1.1"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 6.204l-5.528-0.803-2.472-5.009-2.472 5.009-5.528 0.803 4 3.899-0.944 5.505 4.944-2.599 4.944 2.599-0.944-5.505 4-3.899zM8 11.773l-3.492 1.836 0.667-3.888-2.825-2.753 3.904-0.567 1.746-3.537 1.746 3.537 3.904 0.567-2.825 2.753 0.667 3.888-3.492-1.836z"
      />
    </svg>
     
    <svg
      className="detail__icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      version="1.1"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 6.204l-5.528-0.803-2.472-5.009-2.472 5.009-5.528 0.803 4 3.899-0.944 5.505 4.944-2.599 4.944 2.599-0.944-5.505 4-3.899zM8 11.773l-3.492 1.836 0.667-3.888-2.825-2.753 3.904-0.567 1.746-3.537 1.746 3.537 3.904 0.567-2.825 2.753 0.667 3.888-3.492-1.836z"
      />
    </svg>
     
    <svg
      className="detail__icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      version="1.1"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 6.204l-5.528-0.803-2.472-5.009-2.472 5.009-5.528 0.803 4 3.899-0.944 5.505 4.944-2.599 4.944 2.599-0.944-5.505 4-3.899zM8 11.773l-3.492 1.836 0.667-3.888-2.825-2.753 3.904-0.567 1.746-3.537 1.746 3.537 3.904 0.567-2.825 2.753 0.667 3.888-3.492-1.836z"
      />
    </svg>
  </div>
  <div
    className="movie__info__line"
    data-testid="infoLine"
  >
    <svg
      className="detail__icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm3.3 14.71L11 12.41V7h2v4.59l3.71 3.71-1.42 1.41z"
      />
    </svg>
    <div
      data-testid="infoLength"
    >
      length
    </div>
  </div>
  <div
    className="movie__info__line"
    data-testid="infoLine"
  >
    <svg
      className="detail__icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 2l-5.5 9h11z"
      />
      <circle
        cx="17.5"
        cy="17.5"
        r="4.5"
      />
      <path
        d="M3 13.5h8v8H3z"
      />
    </svg>
    <div
      data-testid="infoGenres"
    >
      Genre1, Genre2, Genre3
    </div>
  </div>
</div>
`;
